bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java

```diff
@@ -44,45 +47,93 @@
 import io.github.hapjava.accessories.HomekitAccessory;
 import io.github.hapjava.server.impl.HomekitRoot;
 import io.github.hapjava.server.impl.HomekitServer;
+import io.github.hapjava.server.impl.crypto.HAPSetupCodeUtils;
 
 /**
  * Provides access to openHAB items via the HomeKit API
  *
  * @author Andy Lintner - Initial contribution
  */
-@Component(service = { Homekit.class }, configurationPid = "org.openhab.homekit", property = {
+@Component(service = { Homekit.class }, configurationPid = HomekitSettings.CONFIG_PID, property = {
         Constants.SERVICE_PID + "=org.openhab.homekit", "port:Integer=9123" })
 @ConfigurableService(category = "io", label = "HomeKit Integration", description_uri = "io:homekit")
 @NonNullByDefault
 public class HomekitImpl implements Homekit {
     private final Logger logger = LoggerFactory.getLogger(HomekitImpl.class);
-    private final NetworkAddressService networkAddressService;
-    private final HomekitChangeListener changeListener;
 
-    private HomekitSettings settings;
+    @Reference
+    protected @NonNullByDefault({}) NetworkAddressService networkAddressService;
+    @Reference
+    protected @NonNullByDefault({}) ConfigurationAdmin configAdmin;
+    @Reference
+    protected @NonNullByDefault({}) ItemRegistry itemRegistry;
+    @Reference
+    protected @NonNullByDefault({}) StorageService storageService;
+    @Reference
+    protected @NonNullByDefault({}) MetadataRegistry metadataRegistry;
+
+    private @NonNullByDefault({}) HomekitAuthInfoImpl authInfo;
+    private @NonNullByDefault({}) HomekitSettings settings;
     private @Nullable InetAddress networkInterface;
     private @Nullable HomekitServer homekitServer;
     private @Nullable HomekitRoot bridge;
-    private final HomekitAuthInfoImpl authInfo;
+    private @NonNullByDefault({}) HomekitChangeListener changeListener;
 
     private final ScheduledExecutorService scheduler = ThreadPoolManager
             .getScheduledPool(ThreadPoolManager.THREAD_POOL_NAME_COMMON);
 
+    public HomekitImpl() {
+    }
+
     @Activate
-    public HomekitImpl(@Reference StorageService storageService, @Reference ItemRegistry itemRegistry,
-            @Reference NetworkAddressService networkAddressService, Map<String, Object> config,
-            @Reference MetadataRegistry metadataRegistry) throws IOException, InvalidAlgorithmParameterException {
-        this.networkAddressService = networkAddressService;
-        this.settings = processConfig(config);
+    protected void activate(Map<String, Object> properties) {
```

Comment:
```suggestion
    public HomekitImpl(@Reference StorageService storageService, @Reference ItemRegistry itemRegistry,
            @Reference NetworkAddressService networkAddressService, @Reference MetadataRegistry metadataRegistry,
            @Reference ConfigurationAdmin configAdmin, Map<String, Object> properties) throws IOException, InvalidAlgorithmParameterException
```

and keep the assignments to the final fields.